src/lib/messenger.ts

```diff
@@ -127,5 +128,15 @@ const postAndListen = (arg: Message): Promise<unknown> =>
 		3000,
 	);
 
-export { post, timeout, postAndListen, generateId };
+const onScroll = () =>
+	postAndListen({ type: 'scroll', value: '' }).then(
+		(result) => result as { top: number; bottom: number },
+	);
+
+const onViewport = () =>
```

Comment:
I don't see any references to onScroll or onViewport? Could you clarify why these are being added here?

Reply:
I think here also (as above) the functions are designed to send messages to the parent frame and listen for responses, they are not directly called in the provided code, and are exported from this messenger file file so intended to be used elsewhere.

Reply:
👍 yes this part of the inital migration but seems latrer the `IntersectionObserver`, which is responsible for detecting when the video element enters the viewport and playing or pausing the video accordingly.